clients/venice-consumer/src/main/java/com/linkedin/venice/client/consumer/RecordChangeEventReadOnlySchemaRepository.java

```diff
@@ -0,0 +1,108 @@
+package com.linkedin.venice.client.consumer;
+
+import com.linkedin.venice.client.change.capture.protocol.RecordChangeEvent;
+import com.linkedin.venice.meta.ReadOnlySchemaRepository;
+import com.linkedin.venice.schema.GeneratedSchemaID;
+import com.linkedin.venice.schema.SchemaEntry;
+import com.linkedin.venice.schema.rmd.RmdSchemaEntry;
+import com.linkedin.venice.schema.writecompute.DerivedSchemaEntry;
+import com.linkedin.venice.serialization.avro.AvroProtocolDefinition;
+import java.util.Arrays;
+import java.util.Collection;
+
+
+/**
+ * TODO: This class is a bit of a hack.  We need to add functionality into the server
+ * to be able to query and retrieve schemas for different view types.  But this functionality isn't
+ * available today.  This class right now operates a shim interface to retrieve the right schema,
+ * but it runs the risk of the protocol not being easily evolvable.
+ */
+public class RecordChangeEventReadOnlySchemaRepository implements ReadOnlySchemaRepository {
+  ReadOnlySchemaRepository internalReadOnlySchemaRepository;
+  private static final SchemaEntry SCHEMA = new SchemaEntry(
+      AvroProtocolDefinition.RECORD_CHANGE_EVENT.getCurrentProtocolVersion(),
+      RecordChangeEvent.getClassSchema());
+
+  RecordChangeEventReadOnlySchemaRepository(ReadOnlySchemaRepository internalReadOnlySchemaRepository) {
+    this.internalReadOnlySchemaRepository = internalReadOnlySchemaRepository;
+  }
+
+  @Override
+  public void refresh() {
+    this.internalReadOnlySchemaRepository.refresh();
+  }
+
+  @Override
+  public void clear() {
+    this.internalReadOnlySchemaRepository.clear();
+  }
+
+  @Override
+  public SchemaEntry getKeySchema(String storeName) {
+    return this.internalReadOnlySchemaRepository.getKeySchema(storeName);
+  }
+
+  @Override
+  public SchemaEntry getValueSchema(String storeName, int id) {
+    return SCHEMA;
+  }
+
+  @Override
+  public boolean hasValueSchema(String storeName, int id) {
+    return true;
+  }
+
+  @Override
+  public int getValueSchemaId(String storeName, String valueSchemaStr) {
+    return AvroProtocolDefinition.RECORD_CHANGE_EVENT.getCurrentProtocolVersion();
+  }
+
+  @Override
+  public Collection<SchemaEntry> getValueSchemas(String storeName) {
+    return Arrays.asList(SCHEMA);
+  }
+
+  @Override
+  public SchemaEntry getSupersetOrLatestValueSchema(String storeName) {
+    return SCHEMA;
+  }
+
+  @Override
+  public SchemaEntry getSupersetSchema(String storeName) {
+    return SCHEMA;
+  }
+
+  @Override
+  public GeneratedSchemaID getDerivedSchemaId(String storeName, String derivedSchemaStr) {
+    return null;
```

Comment:
I am wondering if it is more appropriate to throw UnsupportedOperationException here if this call is unexpected. Same as below 3 API

Reply:
Added it and the tests pass, so you got it!